GitUI/BranchTreePanel/RepoObjectsTree.Designer.cs

```diff
@@ -337,74 +364,151 @@ private void InitializeComponent()
+            this.repoTreePanel.Controls.Add(this.leftPanelToolStrip, 0, 0);
+            this.repoTreePanel.Controls.Add(this.branchSearchPanel, 0, 1);
+            this.repoTreePanel.Controls.Add(this.treeMain, 0, 2);
             this.repoTreePanel.Dock = System.Windows.Forms.DockStyle.Fill;
             this.repoTreePanel.Location = new System.Drawing.Point(0, 0);
             this.repoTreePanel.Margin = new System.Windows.Forms.Padding(0);
             this.repoTreePanel.Name = "repoTreePanel";
-            this.repoTreePanel.RowCount = 2;
+            this.repoTreePanel.RowCount = 3;
+            this.repoTreePanel.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.repoTreePanel.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.repoTreePanel.RowStyles.Add(new System.Windows.Forms.RowStyle());
+            this.repoTreePanel.RowStyles.Add(new System.Windows.Forms.RowStyle(System.Windows.Forms.SizeType.Absolute, 20F));
             this.repoTreePanel.Size = new System.Drawing.Size(300, 350);
             this.repoTreePanel.TabIndex = 4;
             // 
+            // leftPanelToolStrip
+            // 
+            this.leftPanelToolStrip.BackColor = System.Drawing.SystemColors.Control;
+            this.leftPanelToolStrip.ClickThrough = true;
+            this.leftPanelToolStrip.GripStyle = System.Windows.Forms.ToolStripGripStyle.Hidden;
+            this.leftPanelToolStrip.Items.AddRange(new System.Windows.Forms.ToolStripItem[] {
+            this.tsbCollapseAll,
+            this.tsbSettings});
+            this.leftPanelToolStrip.Location = new System.Drawing.Point(0, 0);
+            this.leftPanelToolStrip.Name = "leftPanelToolStrip";
+            this.leftPanelToolStrip.Size = new System.Drawing.Size(300, 25);
+            this.leftPanelToolStrip.TabIndex = 5;
+            this.leftPanelToolStrip.Text = "toolStrip1";
+            // 
+            // tsbCollapseAll
+            // 
+            this.tsbCollapseAll.DisplayStyle = System.Windows.Forms.ToolStripItemDisplayStyle.Image;
+            this.tsbCollapseAll.Image = global::GitUI.Properties.Images.CollapseAll;
+            this.tsbCollapseAll.ImageTransparentColor = System.Drawing.Color.Magenta;
+            this.tsbCollapseAll.Name = "tsbCollapseAll";
+            this.tsbCollapseAll.Size = new System.Drawing.Size(23, 22);
+            this.tsbCollapseAll.Text = "toolStripButton1";
+            this.tsbCollapseAll.Click += new System.EventHandler(this.btnCollapseAll_Click);
+            // 
+            // tsbSettings
+            // 
+            this.tsbSettings.DisplayStyle = System.Windows.Forms.ToolStripItemDisplayStyle.Image;
+            this.tsbSettings.DropDownItems.AddRange(new System.Windows.Forms.ToolStripItem[] {
+            this.tsmiShowBranches,
+            this.tsmiShowRemotes,
+            this.tsmiShowTags,
+            this.tsmiShowSubmodules});
```

Comment:
I don't think we need this dropdown button, suggest removing it and placing the 4 buttons it contains directly on the toolbar